corehq/apps/data_analytics/gir_generator.py

```diff
@@ -0,0 +1,126 @@
+                possible_experience = get_possibly_experienced(domain, monthspan)
+                recently_active = GIRTableGenerator.get_active_recent(domain, monthspan)
+                gir_dict = {
+                    'month': monthspan.startdate,
+                    'domain_name': domain.name,
+                    'country':
+                        ', '.join([unicode(COUNTRIES.get(abbr, abbr)) for abbr in domain.deployment.countries]),
+                    'sector': domain.internal.area,
+                    'subsector': domain.internal.sub_area,
+                    'bu': GIRTableGenerator.get_bu(domain),
+                    'self_service': domain.internal.self_started,
+                    'test_domain': TEST_COUCH_TO_SQL_MAP.get(domain.is_test, NOT_SET),
+                    'start_date': domain.date_created,
+                    'device_id': max_device,
+                    'wam': AMPLIFY_COUCH_TO_SQL_MAP.get(domain.internal.amplifies_workers, NOT_SET),
+                    'pam': AMPLIFY_COUCH_TO_SQL_MAP.get(domain.internal.amplifies_project, NOT_SET),
+                    'wams_current': len(user_tuple.performing & user_tuple.experienced),
+                    'active_users': len(user_tuple.active | user_tuple.sms),
+                    'using_and_performing': len(user_tuple.performing),
+                    'not_performing': len(user_tuple.active - user_tuple.performing),
+                    'inactive_experienced':
+                        len((user_tuple.total - user_tuple.active) & user_tuple.experienced),
+                    'inactive_not_experienced':
+                        len((user_tuple.total - user_tuple.active) - user_tuple.experienced),
+                    'not_experienced': len(user_tuple.performing - user_tuple.experienced),
+                    'not_performing_not_experienced':
+                        len(user_tuple.active - user_tuple.performing - user_tuple.experienced),
+                    'active_ever': len(possible_experience | recently_active),
+                    'possibly_exp': len(possible_experience),
+                    'ever_exp': len(user_tuple.experienced),
+                    'exp_and_active_ever': len(user_tuple.active & user_tuple.experienced),
+                    'active_in_span': len(recently_active)
+                }
+                rows.append(gir_dict)
+        GIRRow.objects.bulk_create(
+            [GIRRow(**gir_row) for gir_row in rows]
+        )
+
+    @staticmethod
+    def classify_users(domain, monthspan):
+        performing_users = []
+        experienced_users = []
+        all_users, users_dict, sms = active_mobile_users(domain, monthspan.startdate, monthspan.enddate)
+        for user in all_users:
+            if users_dict.get(user, 0) > domain.internal.performance_threshold:
+                performing_users.append(user)
+            if len(MALTRow.objects.filter(user_id=user).values('month').distinct()) > \
```

Comment:
it would be more efficient to do this as an aggregate query outside of the loop so you don't hit the DB for every user.